tests/integration/targets/iosxr_bgp_neighbor_address_family/tests/common/merged.yaml

```diff
@@ -0,0 +1,116 @@
+---
+- debug:
+    msg: Start iosxr_bgp_nbr_address_family merged integration tests connection={{ ansible_connection}}
+
+- include_tasks: _remove_config.yaml
+
+- include_tasks: _populate_base_config.yaml
+
+- block:
+    - name: Merge the provided configuration with the exisiting running configuration
+      cisco.iosxr.iosxr_bgp_neighbor_address_family: &id001
+        config:
+          as_number: 65536
+          neighbors:
+            - neighbor: 1.1.1.1
+              address_family:
+                - afi: "ipv4"
+                  safi: "unicast"
+                  multipath: true
+                  default_originate:
+                    set: true
+                  weight: 5
+                  capability_orf_prefix: both
+                  send_multicast_attributes:
+                    set: true
+                  soft_reconfiguration:
+                    inbound:
+                      always: true
+                  send_community_gshut_ebgp:
+                    set: true
+                  send_extended_community_ebgp:
+                    set: true
+                  send_community_ebgp:
+                    set: true
+                  origin_as:
+                    validation:
+                      disable: true
+                  maximum_prefix:
+                    max_limit: 10
+                    threshold_value: 20
+                    restart: 10
+                  next_hop_self:
+                    set: true
+                  as_override:
+                    set: true
+                  allowas_in:
+                    value: 4
+                  bestpath_origin_as_allow_invalid: true
+      register: result
+
+
+    - name: Assert that correct set of commands were generated
+      assert:
+        that:
+          - "{{ merged['commands'] | symmetric_difference(result['commands']) |length\
+            \ == 0 }}"
+
+
+    - name: Assert that after dicts were correctly generated
+      assert:
+        that:
+          - merged['after'] == result['after']
+
+    - name: Merge the provided configuration with the existing running configuration
+        (IDEMPOTENT)
+      cisco.iosxr.iosxr_bgp_neighbor_address_family: *id001
+      register: result
+
+    - name: Assert that the previous task was idempotent
+      assert:
+        that:
+          - result['changed'] == false
+          - result.commands|length == 0
+  always:
+    - include_tasks: _remove_config.yaml
+
+- include_tasks: _populate_vrf_base_config.yaml
+
+- block:
+    - name: Merge the provided configuration with the exisiting running configuration(vrf)
+      cisco.iosxr.iosxr_bgp_neighbor_address_family: &id002
+        config:
+          as_number: 65536
+          vrfs:
+            - vrf: vrf1
+              neighbors:
+                - neighbor: 1.1.1.1
```

Comment:
Since these tests run on an actual appliance, it is suggested to use documentation range IP addresses in tasks as defined in https://tools.ietf.org/html/rfc5737.

Reply:
Ack,made changes